File: sdks/python/src/boxsdk/models/operations/get_retention_policies.py
import dataclasses
from typing import Optional
from enum import Enum
from boxsdk import utils
from ..shared import clienterror as shared_clienterror
from ..shared import retentionpolicies as shared_retentionpolicies

class GetRetentionPoliciesPolicyTypeEnum(str, Enum):
    FINITE = "finite"
    INDEFINITE = "indefinite"


@dataclasses.dataclass
class GetRetentionPoliciesQueryParams:
    created_by_user_id: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'created_by_user_id', 'style': 'form', 'explode': True }})
    fields: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'fields', 'style': 'form', 'explode': False }})
    limit: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'limit', 'style': 'form', 'explode': True }})
    marker: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'marker', 'style': 'form', 'explode': True }})
    policy_name: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'policy_name', 'style': 'form', 'explode': True }})
    policy_type: Optional[GetRetentionPoliciesPolicyTypeEnum] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'policy_type', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetRetentionPoliciesRequest:
    query_params: GetRetentionPoliciesQueryParams = dataclasses.field()
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class GetRetentionPoliciesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    retention_policies: Optional[shared_retentionpolicies.RetentionPolicies] = dataclasses.field(default=None)
    

File: ts-sdk/src/sdk/models/operations/getretentionpolicies.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetRetentionPoliciesPolicyTypeEnum {
    Finite = "finite",
    Indefinite = "indefinite"
}


export class GetRetentionPoliciesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=created_by_user_id" })
  createdByUserId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=marker" })
  marker?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=policy_name" })
  policyName?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=policy_type" })
  policyType?: GetRetentionPoliciesPolicyTypeEnum;
}


export class GetRetentionPoliciesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetRetentionPoliciesQueryParams;
}


export class GetRetentionPoliciesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  retentionPolicies?: shared.RetentionPolicies;

  @SpeakeasyMetadata()
  statusCode: number;
}
